pandas/tests/test_graphics_others.py

```diff
@@ -203,33 +215,42 @@ def test_boxplot_legacy(self):
         _check_plot_works(df.boxplot, return_type='dict')
         _check_plot_works(df.boxplot, column=[
                           'one', 'two'], return_type='dict')
-        _check_plot_works(df.boxplot, column=['one', 'two'], by='indic')
+        with tm.assert_produces_warning(UserWarning):
+            _check_plot_works(df.boxplot, column=['one', 'two'],
+                              by='indic')
```

Comment:
Why should these raise a warning? (it seems they do, as travis does not complain about this one)